search_that_hash/__main__.py

```diff
@@ -72,20 +78,149 @@ def main(**kwargs):
+
+    cracking_handler(config)
 
     exit(0)
 
+
+def cracking_handler(config):
+
+    if config["api"]:
+        coloredlogs.install(level=logging.CRITICAL)
+
+    ### SETTING VARIBLES
+
+    hash_processes = []
+    results = []
+    searcher = cracking.Searcher(config)
+
+    #### STH CRACKING
+
+    sth_results, config = sth.crack(config)
+
+    if not sth_results == False:
+        for result in sth_results.values():
+            if not config["greppable"]:
+                if not config["api"]:
+                    printing.Prettifier.sth_print(
+                        result["Hash"],
+                        result["Plaintext"],
+                        result["Type"],
+                        result["Verified"],
+                    )
+                else:
+                    results.append({result["Hash"]: result["Plaintext"]})
+
+    #### MAIN CRACKING
+
+    for chash, types in config["hashes"].items():
+        if types == []:
+            if not config["greppable"]:
+                if config["api"]:
+                    results.append({chash: "No types found for this hash."})
+                    continue
+                printing.Prettifier.error_print("No types found for this hash.", chash)
+            continue
+
+        hash_processes.append(cracking.Searcher.main(searcher, chash, types))
+
+        #### OUTPUTTING
+
+        chash = list(hash_processes[-1].keys())[0]
+        result = hash_processes[-1][chash]
+
+        if result == None and not config["greppable"]:
+            if config["api"]:
+                results.append({chash: "Could not crack hash"})
+                continue
+            printing.Prettifier.error_print("Could not crack hash.", chash)
+            continue
+
+        if not config["greppable"]:
+            if config["api"]:
+                results.append({chash: result})
+                continue
+            printing.Prettifier.one_print(chash, result)
+
+    if config["greppable"]:
+
+        #### ADDING STH RESULTS
+
+        for hash in hash_processes:
+            base_results = list(hash.values())[0]
+            try:
+                base = sth_results[list(hash.keys())[0]]
+
+                base_results[0].update({"STH_API": base["Plaintext"]})
+                base_results.append(
+                    {"Type": base["Type"], "Verified": base["Verified"]}
+                )
+                """ 
+
+				Firstly this takes the hash from the loop and gets is values
+
+				{hash:[{COMPLETED},{FAILED}]}
+
+				The [0] index is for the main list
+
+				The next [0] index is for the successful dicts
+
+				Then it updates it with STH which:
+
+				Gets all hashes in a list and indexs the hash which its looking for
+
+				Then it takes that plaintext and updates the successfull dict with
+
+				{'STH_API':'Result'}
+				"""
+
+            except:
+                base_results[1].update({"STH_API": "Failed"})
+                base_results.append({"Type": "Unkown", "Verified": "N/A"})
+
```

Comment:
Massive try except blocks suck extremely badly, this is a sign you haven't modularised your code down into individual functions